tests/forms_tests/widget_tests/test_clearablefileinput.py

```diff
@@ -31,6 +32,22 @@ def test_clear_input_renders(self):
             Change: <input type="file" name="myfile" />
             """
         ))
+        self.check_html(self.widget_cls(multiple=True), 'myfile', [FakeFieldFile()], html=(
+            """
+            Currently: <a href="something">something</a>
+            <input type="checkbox" name="myfile-clear" id="myfile-clear_id" />
+            <label for="myfile-clear_id">Clear</label><br />
+            Change: <input multiple type="file" name="myfile" />
+            """
+        ))
+        self.check_html(self.widget_cls(multiple=True), 'myfile', [FakeFieldFile(), FakeFieldFile()], html=(
+            """
+            Currently: 2 files
```

Comment:
Why not showing list of links here? Like this you don't actually know what files you're clearing...

Reply:
That was my first thought too, but it is not how browsers do it for the input itself. UX wise this will always look consistent even if a user selects thousands of files. Murphy's Law, someone will do it ;)